Added Visual Studio build tools (#55629)

cw2 opened the pull request:
Added description of build tools (CMake, Ninja and Python) installation as Visual Studio Individual Components.
Added emphasis (bold) of Visual Studio installer sections and components (replaces previously used different styles of quotes).

Please feel free to suggest any corrections or request any changes...¨

Edit: All three tools are discoverable by the build environment checking mechanism, and were also manually verified via `where` command in the build terminal. The Python VS component installs Python Launcher that places py.exe into the system root directory (automatically included in the global PATH variable).

cw2 committed:
Added Visual Studio build tools (#55629)

Added description of build tools (CMake, Ninja and Python) installation as Visual Studio Individual Components.
Added emphasis (bold) of Visual Studio installer sections and components (replaces previously used different styles of quotes).

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=56234) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=56234) before we can accept your contribution.<br/><br/>:x: cw2 [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=56234)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=56234) it.</sub>

hoyosjs reviewed (approved):
Thanks!

hoyosjs commented on docs/workflow/requirements/windows-requirements.md:
```diff
@@ -22,35 +22,43 @@ git config --system core.longpaths true
+- The build tools (CMake, Ninja and Python) can be downloaded and installed separately (see detailed instructions in the [section below](#build-tools)) or by selecting the following **Individual Components**:
+  - **C++ CMake tools for Windows** (includes Ninja),
+  - **Python 3 32-bit** for building x86 target architecture,
+  - **Python 3 64-bit** for building x64 target architecture. 
```
```suggestion
  - **Python 3 64-bit** for building x64 target architecture.
```

hoyosjs committed:
Update docs/workflow/requirements/windows-requirements.md

mangod9 commented:
please ignore the gc-standalone pipeline (it got accidently enabled for PRs, I have disabled it now). Thx.

jkotas commented on docs/workflow/requirements/windows-requirements.md:
```diff
@@ -22,35 +22,43 @@ git config --system core.longpaths true
+  - **Desktop Development with C++** with all default components.
+- The build tools (CMake, Ninja and Python) can be downloaded and installed separately (see detailed instructions in the [section below](#build-tools)) or by selecting the following **Individual Components**:
+  - **C++ CMake tools for Windows** (includes Ninja),
+  - **Python 3 32-bit** for building x86 target architecture,
```
Is it really required to have 32-bit Python to build for x86 target architecture? Can we recommend 64-bit Python for both instead?

Separately installed 64-bit Python works fine for building both x64 and x86 targets.

cw2 replied:
Good point - unfortunately, I have not explicitly checked this combination. I have verified 64-bit Python is required for building x64 targets; I'll check if it can handle x86 also. Thanks

cw2 replied:
Well, it seems I had some mix-ups with different combinations of the Python components and build target architectures. I am now going to use a dedicated virtual machine to have guaranteed clean environments and thoroughly verify all the combinations (it seems mono libraries x64 build with 64-Python VS component tries to use 32-bit python executable from folder that does not exist).

I'll report the results here.

I apologize for any inconvenience, I should have verified the VS components are working properly before opening the issue 😊 Thank you for your patience and understanding.

cw2 replied:
Removed 32-bit Python component, the 64-bit one works fine for building both x86 and x64 targets.

cw2 committed:
Merge branch 'dotnet:main' into docs/55629-build-tools-vs-components

cw2 committed:
Merge branch 'dotnet:main' into docs/55629-build-tools-vs-components

cw2 committed:
Removed Python 32-bit VS component

Python 64-bit component is enough for building x86 and x64 target architectures.

jkotas reviewed (approved):
Thanks
